lib/libutee/include/pta_device.h

```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2019, Linaro Limited
+ */
+
+#ifndef __PTA_DEVICE_H
+#define __PTA_DEVICE_H
+
+#define PTA_DEVICE_UUID { 0x7011a688, 0xddde, 0x4053, \
+		{ 0xa5, 0xa9, 0x7b, 0x3c, 0x4d, 0xdf, 0x13, 0xb8 } }
+
+/*
+ * PTA_CMD_GET_NUM_DEVICES - Get number of TEE bus devices
+ *
+ * param[0] (out value) - value.a: Number of devices present
+ * param[1] unused
+ * param[2] unused
+ * param[3] unused
+ *
+ * Result:
+ * TEE_SUCCESS - Invoke command success
+ * TEE_ERROR_BAD_PARAMETERS - Incorrect input param
+ */
+#define PTA_CMD_GET_NUM_DEVICES		0x0
+
+/*
+ * PTA_CMD_GET_DEVICES - Get device UUIDs
+ *
+ * param[0] (out memref) - buffer: Array of device UUIDs
+ * param[1] unused
+ * param[2] unused
+ * param[3] unused
+ *
+ * Result:
+ * TEE_SUCCESS - Invoke command success
+ * TEE_ERROR_BAD_PARAMETERS - Incorrect input param
+ * TEE_ERROR_SHORT_BUFFER - Output buffer size less than required
+ */
+#define PTA_CMD_GET_DEVICES		0x1
```

Comment:
How about using a unique command with an in/out size? First call is made with `*size == 0` which returns `TEE_ERROR_SHORT_BUFFER` and sets `*size` to the appropriate value. It is a common pattern.

Reply:
While implementing it, it seems that I can't send `NULL` shm reference to OP-TEE to fetch required size. I get data-abort in OP-TEE here: `core/arch/arm/kernel/pseudo_ta.c +74`. So will keep existing interface only.

Reply:
Sounds like a bug, no?

Reply:
I think OP-TEE basically treats `NULL` shm as `physical address = 0` and tries to get corresponding VA. I am not sure if this should be treated as a bug.

Reply:
But still framework expects memref buffer address to be not `NULL` and if it is `NULL` provides error code.